inst/template/gamma_table.hpp

```diff
@@ -0,0 +1,24 @@
+{{header}}
+#ifndef DUST_DISTR_GAMMA_TABLE_HPP
+#define DUST_DISTR_GAMMA_TABLE_HPP
+
+#ifdef __CUDA_ARCH__
```

Comment:
Shouldn't this be `__NVCC__`? I'm not totally sure, but as it's not compiling a device function when it reaches these I think CUDA_ARCH will be undef, and so these end up `const` rather than `__constant__` in all cases. Which means they will get stored in global memory instead of the constant cache

Reply:
with __NVCC__ I got warnings saying that we couldn't use `__constant__` from the host